nomad/structs/structs.go

```diff
@@ -305,8 +305,9 @@ type EvalAckRequest struct {
 
 // EvalDequeueRequest is used when we want to dequeue an evaluation
 type EvalDequeueRequest struct {
-	Schedulers []string
-	Timeout    time.Duration
+	Schedulers       []string
+	Timeout          time.Duration
+	SchedulerVersion uint16
```

Comment:
It gives us a lot of versions while keeping it low overhead. We will never pass 65536 version bumps.